bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BufferedChannel.java

```diff
@@ -88,8 +90,12 @@ public BufferedChannel(FileChannel fc, int writeCapacity, int readCapacity, long
 
     @Override
     public synchronized void close() throws IOException {
+        if (closed) {
```

Comment:
hmm..not sure why it is needed. Following calls are supposed to be idempotent right?

```
ReferenceCountUtil.safeRelease(writeBuffer);
fileChannel.close();
```

Reply:
@reddycharan safeRelease() is not idempotent.
All it does is suppresses exception and logs it:

```
    public static void safeRelease(Object msg) {
        try {
            release(msg);
        } catch (Throwable t) {
            logger.warn("Failed to release a message: {}", msg, t);
        }
    }
```

The only "benefit" we are getting form it (vs .release()) is that it hides the problem.